functors_task_2/student/include/impl.h

```diff
@@ -65,7 +65,17 @@ class DataBrowser
         // find reader
```

Comment:
please apply safeCall/invokeDataRequest to the getDataType* functions (you should change them in a way that is accepted by these functions), without changing their purpose (so refactor their code)